src/uicomponents/uiapexcard.cpp

```diff
@@ -0,0 +1,348 @@
+    // the apexcard docs say v[0] is a milliseconds-since-epoch value but some configs stick date strings in there
+    // Ruben suggests we could also add a bit of post-processing after data_generator that always returns ecma epoch time numbers
+    switch (v[0].type()) {
+      case nlohmann::detail::value_t::string :
+        {
+          auto from = v[0].get<string>();
+          std::cerr << "from=" << from << std::endl;
+          auto fromt = parse8601(std::istringstream{from});
+          fromtu = fromt.time_since_epoch(); // unix epoch time in milliseconds
+        }
+        break;
+      case nlohmann::detail::value_t::number_integer :
+      case nlohmann::detail::value_t::number_unsigned :
+      case nlohmann::detail::value_t::number_float :
+        fromtu = std::chrono::duration<long, std::milli>(v[0].get<long>());
+        break;
+      default:
+        break;
+    }
+    // if (v[0].type() == json::value_t::
+
+    auto pair = std::make_pair<time_t, double>(fromtu.count() / 1000, v[1].get<double>());
+
+    if (fromtu.count() / 1000 <= now.tv_sec) {
+      nowindex = counter;
+    }
+
+    values.push_back(pair);
+    min = std::min(min, pair.second);
+    max = std::max(max, pair.second);
+    counter++;
+  }
+
+  std::cerr << "min=" << min << ", max=" << max << std::endl;
+  std::cerr << values.size() << std::endl;
+  const auto float_factor = 1000.0; // milli-euros
+
+  if (values.empty()) {
+    values.push_back(std::make_pair(0, 0)); // FIXME: hack to avoid a crash inside LVGL with 0 chart points
+    values.push_back(std::make_pair(3600 * 24 * 365 * 10, 100)); // FIXME: hack to avoid a crash inside LVGL with 0 chart points
+    min = 0;
+    max = 100;
+  }
+  lv_chart_set_point_count(chart, values.size()); // hours
+  lv_chart_set_range(chart, LV_CHART_AXIS_PRIMARY_Y, min * float_factor, max * float_factor); // FIXME/FIXTHEM: lvgl 8.3 docs say it's LV_CHART_AXIS_PRIMARY which is just wrong
+  lv_chart_set_axis_tick(chart, LV_CHART_AXIS_PRIMARY_X, 10, 5, values.size(), 1, true, 40); // major ticks point 10 px down, minor 5. values.size() major ticks, and 1 minor (actually means zero!) in between those. [true] labels on major ticks. 40px for labels.
+  lv_chart_set_axis_tick(chart, LV_CHART_AXIS_PRIMARY_Y, 10, 5, 6, 2, true, 50);
+
+  ser1 = lv_chart_add_series(chart, lv_palette_main(LV_PALETTE_BLUE), LV_CHART_AXIS_PRIMARY_Y);
+  lv_coord_t* ser1_array = lv_chart_get_y_array(chart, ser1);
+
+  for (int i = 0; i < values.size(); i++) {
+    ser1_array[i] = values[i].second * 1000;
+  }
+
+  lv_obj_update_layout(chart); // this makes lv_chart_get_point_pos_by_id work later
+
+  lv_point_t now_coordinates;
+  std::cerr << "nowindex=" << nowindex << std::endl;
+  lv_chart_get_point_pos_by_id(chart, ser1, nowindex, &now_coordinates);
+  std::cerr << "coords(x,y)=" << now_coordinates.x << "," << now_coordinates.y << std::endl;
+
+  lv_obj_t* now_label = createLabel(chart, "<-Now");
```

Comment:
Maybe we can use something like LV_SYMBOL_LEFT here?

Reply:
I tried that but it makes the label's position even worse on my test data (which clearly is a positioning problem)